src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -282,6 +290,97 @@ private IEnumerable<BlittableJsonReaderObject> ReadArray(INewDocumentActions act
+            try
+            {
+                while (true)
+                {
+                    if (UnmanagedJsonParserHelper.Read(_stream, _parser, _state, _buffer) == false)
+                        ThrowInvalidJson("Unexpected end of json while reading legacy attachments");
+
+                    if (_state.CurrentTokenType == JsonParserToken.EndArray)
+                        break;
+
+                    if (actions != null)
+                    {
+                        var oldContext = context;
+                        context = actions.GetContextForNewDocument();
+                        if (oldContext != context)
+                        {
+                            builder.Dispose();
+                            builder = CreateBuilder(context, modifier);
+                        }
+                    }
+                    builder.Renew("import/object", BlittableJsonDocumentBuilder.UsageMode.ToDisk);
+
+                    ReadObject(builder);
+
+                    var data = builder.CreateReader();
+                    builder.Reset();
+
+                    var attachment = new DocumentItem.AttachmentStream
+                    {
+                        Stream = actions.GetTempStream()
+                    };
+
+                    var attachmentInfo = ProcessLegacyAttachment(context, data, ref attachment);
+                    var dummyDoc = new DocumentItem()
+                    {
+                        Document = new Document()
+                        {
+                            Data = WriteDummyDocumentForAttachment(context, attachmentInfo),//TODO:this is wrong i need to generate a dummy document just like in the DR tool
+                            Id = attachmentInfo.Id,
+                            ChangeVector = string.Empty,
+                            Flags = DocumentFlags.HasAttachments,
+                            NonPersistentFlags = NonPersistentDocumentFlags.FromSmuggler
+                        }
+                    };
+                    dummyDoc.Attachments = new List<DocumentItem.AttachmentStream>();
+                    dummyDoc.Attachments.Add(attachment);
+                    yield return dummyDoc;
+                }
+            }
+            finally
+            {
+                builder.Dispose();
+            }
+
+        }
+
+        private BlittableJsonReaderObject WriteDummyDocumentForAttachment(DocumentsOperationContext context, LegacyAttachmentDetails details)
+        {
+            var attachment = new DynamicJsonValue
+            {
+                ["Name"] = details.Key,
+                ["Hash"] = details.Hash,
+                ["ContentType"] = string.Empty,
+                ["size"] = details.Size,
+            };
+            var attachmets = new DynamicJsonArray();
+            attachmets.Add(attachment);
+            var metadata = new DynamicJsonValue
+            {
+                [Constants.Documents.Metadata.Collection] = "@empty",
```

Comment:
Probably `@empty` is not a good idea

Reply:
@attachments?

Reply:
Just `Attachments`, I think, or even `Files` if this is from RavenFS

Reply:
+1 for `@files`.

Reply:
again this wasn't implemented for ravenfs, we need to support attachments from 3.x so I would go with Attachments

Reply:
Ok, than use `@attachments` as the user might have their own `Attachments` collection, so we insert implicit collection with the `@` prefix.

Reply:
okay